xarray/backends/api.py

```diff
@@ -1040,11 +1078,16 @@ def to_zarr(dataset, store=None, mode='w-', synchronizer=None, group=None,
     _validate_dataset_names(dataset)
     _validate_attrs(dataset)
 
+    if mode == "a":
```

Comment:
Can we raise an error if `encoding` was passed explicitly into `to_zarr()` and specifies encoding for any existing variables? I.e., `ds.to_arr(..., mode='a', encoding={'existing_variable': ....})`. I think this would always indicate a programming mistake, given that we throw away these variable encodings anyways.